sdk/src/metrics/meter.cc

```diff
@@ -213,18 +212,15 @@ std::unique_ptr<WritableMetricStorage> Meter::RegisterMetricStorage(
 /** collect metrics across all the meters **/
 bool Meter::Collect(CollectorHandle *collector,
                     opentelemetry::common::SystemTimestamp collect_ts,
-                    nostd::function_ref<bool(MetricData &)> callback) noexcept
+                    nostd::function_ref<bool(MetricData &&)> callback) noexcept
 {
   std::vector<MetricData> data;
   for (auto &metric_storage : storage_registry_)
   {
     // TBD - this needs to be asynchronous
-    metric_storage.second->Collect(collector, meter_context_->GetCollectors(),
-                                   meter_context_->GetSDKStartTime(), collect_ts,
-                                   [&callback](MetricData &metric_data) {
-                                     callback(metric_data);
-                                     return true;
-                                   });
+    metric_storage.second->Collect(
+        collector, meter_context_->GetCollectors(), meter_context_->GetSDKStartTime(), collect_ts,
+        [&callback](MetricData &&metric_data) { return callback(std::move(metric_data)); });
```

Comment:
Would returning false from one callback stop the collection process?

Reply:
There can be multiple instruments configured for a Meter. This will fail the collection for the instrument for which the callback failed, and the rest of the instrument collection would continue.